python/ray/tune/logger/json.py

```diff
@@ -94,6 +94,10 @@ def log_trial_start(self, trial: "Trial"):
         # Make sure logdir exists
         trial.init_local_path()
         local_file = os.path.join(trial.local_path, EXPR_RESULT_FILE)
+
+        # Resume the file from remote storage.
+        self._restore_from_remote(EXPR_RESULT_FILE, trial)
```

Comment:
@justinvyu implemented your idea here. One question--- would this be a performance concern if we have to do this remote call on each trial start? Is there any way we can skip this restore call entirely if we know this is the first time the trial ran?

Reply:
Maybe we can just check if `trial.checkpoint` exists. If so, try pulling the file.

Otherwise it's just starting from scratch, so let's just write a new metrics file at that point.

Reply:
I like it!

Reply:
I ended up adding a TODO here instead, since `trial.checkpoint` wasn't initially populated from what I could tell. Not sure if that is expected from restore

Reply:
Hmm. it should be there, what's the test script?

Reply:
Here's the script I use (just Ctrl-C it after a while, then comment in the restore path):

```
from ray import tune
import glob
import shutil
import pytest
import time
from ray.air.config import RunConfig
from ray.air import session
from ray.train._checkpoint import Checkpoint
import os

NUM_MODELS = 2000


def train_model(config):
    time.sleep(1)
    print("Training model woohoo!")

    # Import model libraries, etc...
    # Load data and train model code here...
    with open("random_artifact.txt", "w") as f:
        f.write("artifact data hi there")

    # Return final stats. You can also return intermediate progress
    # using ray.air.session.report() if needed.
    # To return your model, you could write it to storage and return its
    # URI in this dict, or return it as a Tune Checkpoint:
    # https://docs.ray.io/en/latest/tune/tutorials/tune-checkpoints.html
    for i in range(5):
        with open("/tmp/data/checkpoint.data", "w") as f:
            f.write(f"Hello world {i}")
        time.sleep(3)
        session.report(
            {"score": 2.0, "epoch": i},
            checkpoint=Checkpoint.from_directory("/tmp/data"),
        )


# Start a Tune run and print the best result.
rc = RunConfig(storage_path="/tmp/tune_out")

trial_space = {
    # This is an example parameter. You could replace it with filesystem paths,
    # model types, or even full nested Python dicts of model configurations, etc.,
    # that enumerate the set of trials to run.
    "model_id": tune.grid_search(["model_{}".format(i) for i in range(NUM_MODELS)])
}

tuner = tune.Tuner(train_model, param_space=trial_space, run_config=rc)
#tuner = tune.Tuner.restore(
#    glob.glob("/tmp/tune_out/train_model_*")[0],
#    trainable=train_model)
tuner.fit()
```

Reply:
I'll try it out in a bit.